base/indices.jl

```diff
@@ -242,7 +262,28 @@ size(S::Slice) = first(S.indices) == 1 ? (length(S.indices),) : errmsg(S)
 length(S::Slice) = first(S.indices) == 1 ? length(S.indices) : errmsg(S)
 unsafe_length(S::Slice) = first(S.indices) == 1 ? unsafe_length(S.indices) : errmsg(S)
 getindex(S::Slice, i::Int) = (@_inline_meta; @boundscheck checkbounds(S, i); i)
+function getindex(r::Slice, s::AbstractUnitRange{<:Integer})
+    @_inline_meta
+    @boundscheck checkbounds(r, s)
+    s
+end
 show(io::IO, r::Slice) = print(io, "Base.Slice(", r.indices, ")")
 start(S::Slice) = start(S.indices)
 next(S::Slice, s) = next(S.indices, s)
 done(S::Slice, s) = done(S.indices, s)
+intersect{I<:AbstractUnitRange{Int}}(r::Slice{I}, s::Slice{I}) =
+    Slice(convert(I, max(first(r), first(s)):min(last(r), last(s))))
+intersect(r::Slice, s::Slice) =
+    Slice(max(first(r), first(s)):min(last(r), last(s)))
+reverse(r::Slice) = error("reverse is not supported for Base.Slice")
```

Comment:
The general `AbstractArray` reverse would throw an error for an offset array, but `OrdinalRange` just always creates a 1-indexed array.